src/aggregation/bucket/histogram.rs

```diff
@@ -0,0 +1,15 @@
+use crate::{aggregation::agg_result::BucketAggregationResult, collector::SegmentCollector};
+
+struct Histogram {}
+
+impl SegmentCollector for Histogram {
+    type Fruit = BucketAggregationResult;
```

Comment:
Can we keep the Fruit specific to each type at this level, and have type conversion into BucketAggregationResult?

Reply:
Good point, this is what I do in the metric aggregation already, but for the bucket it is a tree, e.g. something like
- bucketL1 Color Key Green docs: 50
   - bucketL2 Year Key 1900 docs: 20
   - bucketL2 Year Key 2000 docs: 30
- bucketL1 Color Key Red docs: 20
   - bucketL2 Year Key 1900 docs: 5
   - bucketL2 Year Key 2000 docs: 15

I think the alternative would be that the aggregation is called for every subbucket, in that case twice (Green and Red) and the tree is not know in the aggregation itself. The handling would then be on the execution side, which is missing currently, so I'm not sure how good that approach is.